Training_Preparation.md

```diff
@@ -0,0 +1,275 @@
+
+By the way, do you remember the HTTP-header named "edc-bpn", which we were looking at during task 1? Is it needed here too? 
+
+
+#### g) Querying a Specific Shell via EDC Transfer
+This step corresponds to the step from Task 1 d, but using an EDC transfer. Note that we can use the same contract as the basis for all requests. We can even use the same transfer that we requested in Task 3 d. Only the token may need to be renewed if it has expired.
+
+
+#### h) Catalog Query for the PartTypeApi Asset
+
+
+First, let's look again at the response with the shell descriptor from the previous query. It is recommended to copy and paste the content into the code editor. Then, if we search through the "submodelDescriptors" array, we find an entry that contains the following: "urn:samm:io.catenax.part_type_information:1.0.0#PartTypeInformation".
+
+
+Let's take a closer look at this section. Within this JSON object, there is also an "endpoints" array. The entries "href" and "subprotocolBody" are particularly interesting. Evaluating this information allows us to know where and how we can access the submodel described in this shell descriptor.
+
+
+We want to spare you the trouble of copying these data individually and transferring them to other request bodies.
+
+
+If we now look at the current request body, we see that we use the string "urn:samm:io.catenax.part_type_information:1.0.0#PartTypeInformation" as a filter criterion to start another catalog query.
+
+#### i) Negotiation for a Contract over the PartType Asset
+
+
+Business as usual :)
+
+
+#### j) Querying the Negotiation Status
+As a response, we receive the contract ID.
+
+
+#### k) Requesting a Pull Transfer
+We provide the contract ID.
+
+
+#### l) Requesting an EDR
+As above.
+
+
+#### m) Pull Data from PartType-API
+
+
+We now see what information the PartType-API provides regarding this material. It's also interesting to note that the PURIS backend internally handled the response. If we take a quick look at the logs of our PURIS backend containers, we can see that the PartTypeInformationController has processed a request.
+
+
+### Task 4: Creating a New Sample Asset on Our EDC
+
+
+#### a) Asset Registration
+
+
+With this request, we create a new asset on the EDC of the supplier partner.
+
+
+Let's briefly look at the request body. We create a new fictitious API asset there, which we have given the identifier "urn:samm:io.catenax.sample_information:1.0.0#SampleInformation".
+
+
+Since we do not have an endpoint in the PURIS backend that can serve this asset, we simply use any URL in the "baseUrl" field that returns a list of artificial user names.
+
+
+#### b) Creating a Contract Definition
+
+
+Here, we reference the newly created asset in the request body.
+
+
+Additionally, we reference in the "accessPolicy" and "contractPolicy" fields two policies that our PURIS backend already created on the EDC during its boot process.
```

Comment:
Append: In case you're further interested, checkout the [PURIS documentation](https://github.com/eclipse-tractusx/puris/blob/main/docs/architecture/08_concepts.md#data-sovereignty) explaining the policies used and the data sovereignty concept.